source/exe/win32/platform_checks.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include "envoy/filesystem/filesystem.h"
+
+namespace Envoy {
+
+// Check system wide configurations such as user level
+// file watches, file limit permissions etc...
+static void check_platform_settings(Filesystem::Instance&) {
+  std::cerr << "platform specific checks are not implemented for win32 - please ensure that "
```

Comment:
I'm not a fan of having Envoy spew this out unconditionally...maybe leave a TODO for the windows porting folks here though?